src/provider/common/Injectables.ts

```diff
@@ -22,7 +22,14 @@ enum Injectable {
     WEB_WINDOW_FACTORY
 }
 
-type InjectableMap = {
+/**
+ * A map type to enforce that we specify a value for every injectable
+ */
+export type InjectableMap<V = object> = {
+    [K in keyof typeof Injectable]: K extends string ? V : never;
+}
+
+type InjectableSelfMap = {
```

Comment:
we could actually simplify this whole type mess and and extra key filter/reduce process and get rid of `Inject` with;

```
enum Injectables {
    API_HANDLER,
    CLIENT_EVENT_CONTROLLER,
    CLIENT_REGISTRY
    ...
};

type InjectableMap<V = object> = {
    [K in Injectables]: V;
}
```

then all we need to do is to export the Injectable enum and instantiate your InjectableMap with:
```
const Bindings: InjectableMap = {
    [Injectables.API_HANDLER]: APIHandler,
    [Injectables.CLIENT_EVENT_CONTROLLER]: ClientEventController,
    [Injectables.CLIENT_REGISTRY]: ClientRegistry,
    ...
}
```

Reply:
I guess this is because strings work well with @inject?

Reply:
Yeah, think this was driven by needing `Inject.WHATEVER` to be a string to work with `@inject`. And once we have that, I think it makes sense to enforce that that's what we use rather than the enum